src/Sulu/Bundle/MediaBundle/Entity/FileVersionMetaRepository.php

```diff
@@ -33,7 +32,8 @@ public function findLatestWithoutSecurity()
                 AccessControl::class,
                 'accessControl',
                 'WITH',
-                'accessControl.entityClass = :entityClass AND accessControl.entityId = collection.id'
+                'accessControl.entityClass = :entityClass '
+                . 'AND accessControl.entityId = CAST(collection.id AS CHAR(36))'
```

Comment:
the `CAST` function of the `oroinc/doctrine-extensions` package supports `string` as type. maybe we could use that instead of hardcoding a length here?

Reply:
think the `CAST` function will use `char` on mysql and `varchar` on postgres when using `string` 🙂 

- https://github.com/oroinc/doctrine-extensions/blob/master/src/Oro/ORM/Query/AST/Platform/Functions/Mysql/Cast.php
- https://github.com/oroinc/doctrine-extensions/blob/master/src/Oro/ORM/Query/AST/Platform/Functions/Postgresql/Cast.php